packages/lumx-react/src/components/fieldset/Fieldset.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { ReactElement } from 'react';
+
+import classNames from 'classnames';
+
+import { CSS_PREFIX } from '@lumx/core/js/constants';
+import { Size, Theme } from '@lumx/react';
+import { COMPONENT_PREFIX } from '@lumx/react/constants';
+import { IGenericProps, getRootClassName, handleBasicClasses } from '@lumx/react/utils';
+
+/**
+ * Defines the props of the component.
+ */
+interface IFieldsetProps extends IGenericProps {
+    legend?: ReactElement | string;
+    topSpacing?: Size;
+    bottomSpacing?: Size;
+    theme?: Theme;
+}
+
+interface IDefaultPropsType extends Partial<IFieldsetProps> {}
+const FIELDSET_DEFAULT_PROPS: IDefaultPropsType = {
+    bottomSpacing: Size.big,
+    theme: Theme.light,
+    topSpacing: Size.huge,
+};
+
+const COMPONENT_NAME = `${COMPONENT_PREFIX}Fieldset`;
+
+/**
+ * The default class name and classes prefix for this component.
+ *
+ */
+const CLASSNAME = getRootClassName(COMPONENT_NAME);
+
+const Fieldset: React.FC<IFieldsetProps> = ({
+    bottomSpacing = FIELDSET_DEFAULT_PROPS.bottomSpacing,
+    className = '',
+    children,
+    legend,
+    theme = FIELDSET_DEFAULT_PROPS.theme,
+    topSpacing = FIELDSET_DEFAULT_PROPS.topSpacing,
+    ...props
+}: IFieldsetProps): React.ReactElement => (
+    <fieldset className={classNames(className, CLASSNAME, handleBasicClasses({ prefix: CLASSNAME, theme }))} {...props}>
+        {legend && (
+            <legend
+                className={classNames(
+                    `${CSS_PREFIX}-spacing-margin-bottom-${bottomSpacing}`,
```

Comment:
Seems that `bottomSpacing` and `topSpacing` are related to the fact that the `legend` is present or not right ? Maybe we could add some wording, or change the `topSpacing` and `bottomSpacing` to reflect that they are indeed related to the legend ? From the get go, I thought they were related to the spacing for the fieldset, rather than the spacing for the legend. What do you think @radiuso ?

Reply:
I think I will delete the topSpacing prop since we can use the class "lumx-spacing-margin" on the Fieldset component itself.

Regarding this, it could be the responsibility of the first input to have a class "lumx-spacing-margin".

What do you think ?